### DISCUSSION TITLE
external database

### DISCUSSION THREAD

**Discussion by mtnbiker777** [photoprism/photoprism#1069] [General]
If this has been asked before I apologize, I looked and don't see it. I would like photoprism to use an existing container running MySQL. I've tried changing the docker-compose to point to that container but it didn't work. I've noticed photoprism created it's own network, and that might be part of the issue. The database never gets created, and when I try and logon to the front end I get invalid credentials using the admin account and password set in the docker-compose. If you can shove me in the right direction I'd appreciate it

**Comment by lastzero** [comment]
Docker automatically creates a new network, you may set an external network as described in their docs:

https://docs.docker.com/compose/networking/#use-a-pre-existing-network

For using a database running on your host, see #1059.

**Reply by mtnbiker777** [reply]
Thanks that was helpful, I apologize for the stupidity of the docker question, I've used docker but not compose. I was able to get photoprism on the same network as the mysql database so that's a start. I removed all things maria from the compose. I manually created an account that is DBA on the database server that is photoprism with the password in the docker compose and updated the line

PHOTOPRISM_DATABASE_SERVER: "172.18.0.10:3306"

which is my existing server. The database is not being created so i can only assume I have something wrong. Do I need to manually create the database?

**Reply by lastzero** [reply]
Yes, photoprism doesn't assume root permissions so you need to setup permissions and create the schema on your own as admin / root.

**Reply by mtnbiker777** [reply]
I manually created the schema, gave the photoprism account full rights to the schema and no tables or anything have been created. Is there any debugging I can turn on to see database errors? I turned on debugging in the compose ...

**Reply by lastzero** [reply]
We have a debug mode with additional logs, also Docker has logs

**Reply by mtnbiker777** [reply]
Well it seems my issue is that I thought I had managed to get the docker compose to use the existing network but it didn't it still had created it's own network and couldn't communicate with the database contained on the same host. I get it, this is a docker compose issue NOT a photoprism issue.

**Reply by mtnbiker777** [reply]
I gave up on the docker compose and got it working fine. Thanks for the helps.
